<?xml version="1.0" encoding="UTF-8"?> 
<launch>
  <arg name="robot_number" value="1"/>
  <arg name="do_ranging" value="0"/>
  <arg name="left_tag_x" default="0" />
  <arg name="left_tag_y" default="0.255" />
  <arg name="right_tag_x" default="0" />
  <arg name="right_tag_y" default="-0.255" />

  <arg name="robot_name" default="turtle_$(arg robot_number)" />
  <arg name="prefix_tf" default="$(arg robot_name)_tf" />
  
  <arg name="link" default="$(arg prefix_tf)/plate_top_link" />

  <param name="tf_prefix" value="$(arg prefix_tf)"/>

  <group ns="$(arg robot_name)">
    <node pkg="amr_x_communication" type="uwb_node.py" name="localization_node" output="screen">
      <param name="robot_number" value="$(arg robot_number)"/>
      <param name="serial_port" value="/dev/ttyACM0"/>
      <param name="frequency" value="20"/>
      <param name="do_ranging" value="$(arg do_ranging)"/>

      <param name="left_tag_pos_x" value="$(arg left_tag_x)"/>
      <param name="left_tag_pos_y" value="$(arg left_tag_y)"/>
      <param name="right_tag_pos_x" value="$(arg right_tag_x)"/>
      <param name="right_tag_pos_y" value="$(arg right_tag_y)"/>
      <param name="tf_prefix" value="$(arg prefix_tf)"/>
    
      <param name="loc_dis" value="5"/> <!--Distance to start localization-->
      <param name="com_dis" value="2"/> <!--Distance to start communication-->
      <param name="link" value="$(arg link)"/> 
     
      <!--Filter settings for localization-->
      <param name="R" value="50"/>
      <param name="alpha" value="1.07"/>
      <param name="noise" value="0.001"/>
      <param name="protocol" value="1"/>

      <!--Communication settings-->
      <param name="tx_topic" value="odom"/>
      <param name="rx_topic" value="uwb_rx"/>
    </node>
    
    <node name="left_tag_link" pkg="tf" type="static_transform_publisher" args="$(arg left_tag_x) $(arg left_tag_y) 0 0 0 0 $(arg link) $(arg prefix_tf)/left_tag_link 10"/>
    <node name="right_tag_link" pkg="tf" type="static_transform_publisher" args="$(arg right_tag_x) $(arg right_tag_y) 0 0 0 0 $(arg link) $(arg prefix_tf)/right_tag_link 10"/>
    
    <include file="$(find amr_x_communication)/launch/minimal.launch">
      <arg name="prefix_tf" value="$(arg prefix_tf)"/>
    </include>
  </group>

  <!--node pkg="robot_localization" type="ekf_localization_node" name="ekf_localization" clear_params="true" output="screen">
    <rosparam command="load" file="$(find amr_x_communication)/params/ekf_template.yaml" />
  </node-->
</launch>
